### PR TITLE
feat: add folder sharing, fix "shared with me" displaying (#291)

### PR DESCRIPTION
**Pull request by IlyaBondar**
### Description
- Issue #291

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Deployment status: success

File: src/store/conversations/conversations.selectors.ts [hunk 1]
```diff
@@ -63,30 +64,38 @@ export const selectFilteredFolders = createSelector(
...
     );
     const folderIds = filteredConversations // direct parent folders
       .map((c) => c.folderId)
       .filter((fid) => fid);
-    // include empty folders only if not search
-    if (includeEmptyFolders && !searchTerm?.trim().length) {
-      folderIds.push(...emptyFolderIds);
+
+    if (!searchTerm?.trim().length) {
+      const markedFolderIds = folders
```
Comment (mikitabut) [inline src/store/conversations/conversations.selectors.ts:90]: nit: reduce will be better to use here

File: src/types/search.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+
 export enum SearchFilters {
   None = 0,
   SharedByMe = 1 << 0,
   PublishedByMe = 1 << 1,
 }
+
+export type EntityFilter<T> = (item: T) => boolean;
+export interface EntityFilters {
+  rootFilter: EntityFilter<ShareEntity>; // filter root level folders and items e.g. "Shared with me"
```
Comment (mikitabut) [inline src/types/search.ts]: Do we really need this differentiation? And how it will work? Could you provide more examples?

**Review by mikitabut — APPROVED** [review]

**Pull request merged**